### ISSUE TITLE
Improve 'prefer-immediate-return': ignore when returned variable has a declared type

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/eslint-plugin-sonarjs#274]
In 'prefer-immediate-return' when returned variable declaration has a declared type it could be sometimes error-prone to drop this typed declaration
See detailed example https://community.sonarsource.com/t/immediately-returned-variable-rule-conflicts-with-typescript-excess-property-checking/30870

```
function foo() {
  let x: SomeType = ...; // do not report if declaration has a type
  return x; 
}
```

**Comment by WernerClemens** [comment]
I have the same issue.
